src/main/java/org/jenkinsci/plugins/workflow/job/WorkflowRun.java

```diff
@@ -152,6 +160,21 @@ public String url() {
     private transient boolean allowKill;
 
     /**
+     * Cumulative list of people who contributed to the build problem.
+     *
+     * <p>
+     * This is a list of {@link User#getId() user ids} who made a change
+     * since the last non-broken build. Can be null (which should be
+     * treated like empty set), because of the compatibility.
+     *
+     * <p>
+     * This field is semi-final --- once set the value will never be modified.
+     *
+     * @since 1.137
+     */
+    private volatile Set<String> culprits;
```

Comment:
The top-level `if` switch in the upstream PR seems very odd to me, but I cannot put my finger on what looks wrong. Probably the mixing of a method to _compute_ culprits with a method to _retrieve_ culprits.

At any rate, you need to document upstream that this logic only works when the list of culprit IDs is fixed at the time of build completion—it apparently cannot be computed on demand.